src/h5web/explorer/utils.ts

```diff
@@ -87,17 +88,32 @@ export function buildTree(
   });
 }
 
+function findRoot(entity: MyHDF5Group): MyHDF5Metadata {
+  const { parent } = entity;
+  return parent ? findRoot(parent) : entity;
+}
+
 export function getEntityAtPath(
-  entity: MyHDF5Entity,
-  path: number[]
-): MyHDF5Entity {
-  if (path.length === 0 || !isGroup(entity)) {
-    return entity;
+  baseGroup: MyHDF5Group,
+  path: string,
+  allowSelf = true
```

Comment:
I found it more "natural" for this function to return the entity itself when the path is `''` (or when the path is `'/'` and the entity is the root entity). However, this behaviour is problematic in `findNxData`, when following an empty default attribute (or `'/'` on the root), because it leads to infinite recursion. I chose to add this attribute to make the behaviour explicit.

Reply:
Looks good :+1: